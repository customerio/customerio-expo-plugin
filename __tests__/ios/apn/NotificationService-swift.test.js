const { testAppPath, createPartialSnapshot } = require("../../utils");
const fs = require("fs-extra");
const path = require("path");

const testProjectPath = testAppPath();
const iosPath = path.join(testProjectPath, "ios");
const notificationServicePath = path.join(iosPath, "NotificationService/NotificationService.swift");

describe('APN NotificationService.swift', () => {
  test("Plugin creates NotificationService.swift with required methods", async () => {
    const content = await fs.readFile(notificationServicePath, "utf8");
    
    // Check for required elements rather than exact snapshot
    const requiredElements = [
      'didReceive',
      'withContentHandler',
      'serviceExtensionTimeWillExpire'
    ];
    
    requiredElements.forEach(element => {
      expect(content).toContain(element);
    });
  });
  
  test("NotificationService.swift has expected APN configuration", async () => {
    const content = await fs.readFile(notificationServicePath, "utf8");
    
    // Check for APN vs FCM implementation details
    expect(content).toContain('MessagingPush');
    expect(content).toContain('CioMessagingPushAPN');
  });
});
