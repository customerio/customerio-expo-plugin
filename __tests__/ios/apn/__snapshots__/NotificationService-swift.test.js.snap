// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin creates expected NotificationService.swift 1`] = `
"import Foundation
import UserNotifications
import CioMessagingPushAPN

@objc
public class NotificationServiceCioManager : NSObject {
  
  public override init() {}
  
  @objc(didReceive:withContentHandler:)
  public func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
    MessagingPushAPN.initializeForExtension(
      withConfig: MessagingPushConfigBuilder(cdpApiKey: Env.customerIOCdpApiKey)
        .region(Env.customerIORegion)
        .build()
    )
    
    MessagingPush.shared.didReceive(request, withContentHandler: contentHandler)
  }
  
  @objc(serviceExtensionTimeWillExpire)
  public func serviceExtensionTimeWillExpire() {
    MessagingPush.shared.serviceExtensionTimeWillExpire()
  }
}
"
`;
